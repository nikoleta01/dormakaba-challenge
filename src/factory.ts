import { ConnectionStatus } from './models/ConnectionStatus';
import { ConnectionType } from './models/ConnectionType';
import { Door } from './models/Door';
import { faker } from '@faker-js/faker';

export function buildDoor(door?: Partial<Door>) {
  return {
    id: faker.string.uuid(),
    name: faker.lorem.word(),
    buildingName: faker.lorem.words(2),
    connectionType: faker.helpers.enumValue(ConnectionType),
    connectionStatus: faker.helpers.enumValue(ConnectionStatus),
    lastConnectionStatusUpdate: faker.date.past().toISOString(),
    apartmentName: `Apartment ${faker.number.int(1000)}`,
    ...door,
  };
}
