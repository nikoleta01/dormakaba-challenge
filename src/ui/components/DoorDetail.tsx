import Typography from '@mui/material/Typography';
import { Door } from '@/models/Door';
import { DetailPageContainer } from '@/ui/layout/DetailPageContainer';
import { DetailPageItem } from '@/ui/layout/DetailPageItem';

interface DoorDetailProps {
  door: Door;
}

export function DoorDetail({ door }: DoorDetailProps) {
  return (
    <DetailPageContainer>
      <DetailPageItem label="ID">
        <Typography>{door.id}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Building">
        <Typography>{door.buildingName}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Connection type">
        <Typography>{door.connectionType}</Typography>
      </DetailPageItem>
      <DetailPageItem label="Connection status">
        <Typography
          color={
            door.connectionStatus === 'online' ? 'success.main' : 'error.main'
          }
        >
          {door.connectionStatus}
        </Typography>
      </DetailPageItem>
      <DetailPageItem label="Apartment name">
        <Typography>{door.apartmentName}</Typography>
      </DetailPageItem>
    </DetailPageContainer>
  );
}
