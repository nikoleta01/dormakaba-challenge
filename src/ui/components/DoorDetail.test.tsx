import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DoorDetail } from './DoorDetail';
import { buildDoor } from '@/factory';
import { Door } from '@/models/Door';

function renderDoorDetail(door: Door) {
  return render(<DoorDetail door={door} />);
}

describe('DoorDetail', () => {
  it('should render door details', () => {
    const door = buildDoor();
    renderDoorDetail(door);
    expect(screen.getByText('ID')).toBeInTheDocument();
    expect(screen.getByText(door.id)).toBeInTheDocument();

    expect(screen.getByText('Building')).toBeInTheDocument();
    expect(screen.getByText(door.buildingName)).toBeInTheDocument();

    expect(screen.getByText('Connection type')).toBeInTheDocument();
    expect(screen.getByText(door.connectionType)).toBeInTheDocument();

    expect(screen.getByText('Connection status')).toBeInTheDocument();
    expect(screen.getByText(door.connectionStatus)).toBeInTheDocument();

    expect(screen.getByText('Apartment name')).toBeInTheDocument();
    expect(screen.getByText(door.apartmentName)).toBeInTheDocument();
  });
  it('applies green color to connection status online', () => {
    const door = buildDoor({
      connectionStatus: 'online',
    });
    renderDoorDetail(door);

    const connectionStatusElement = screen.getByText(door.connectionStatus);
    expect(connectionStatusElement).toHaveStyle('color: rgb(46, 125, 50)');
  });
  it('applies red color to connection status offline', () => {
    const door = buildDoor({
      connectionStatus: 'offline',
    });
    renderDoorDetail(door);

    const connectionStatusElement = screen.getByText(door.connectionStatus);
    expect(connectionStatusElement).toHaveStyle('color: rgb(211, 47, 47)');
  });
});
