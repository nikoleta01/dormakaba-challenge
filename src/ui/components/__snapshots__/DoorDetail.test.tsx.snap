// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoorDetail should render correctly 1`] = `
<div
  class="MuiStack-root css-1sazv7p-MuiStack-root"
>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      <strong>
        ID
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      63f637c9f3c48a124616044b
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      <strong>
        Building
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      Bahnhofstrasse 10A
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      <strong>
        Connection type
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      wired
    </p>
  </div>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      <strong>
        Connection status
      </strong>
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 css-wl91uy-MuiTypography-root"
    >
      online
    </p>
  </div>
</div>
`;
